src/Bridge/Doctrine/Orm/Filter/DateFilter.php

```diff
@@ -167,7 +167,7 @@ protected function filterProperty(string $property, $values, QueryBuilder $query
     protected function addWhere(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $alias, string $field, string $operator, string $value, string $nullManagement = null, $type = null)
     {
         try {
-            $value = false !== strpos('_immutable', $type) ? new \DateTime($value) : new \DateTimeImmutable($value);
+            $value = false !== strpos($type, '_immutable') ? new \DateTimeImmutable($value) : new \DateTime($value);
```

Comment:
Can you invert the test btw? It will be easier to read: `false === strpos($type, '_immutable') ? new \DateTime($value) : new \DateTimeImmutable($value)`?